### ISSUE TITLE
calendarPlot wind direction and speed

### ISSUE THREAD

**Issue by wasdzxc2007** [davidcarslaw/openair#130]
I was trying to plot daily ozone concentration vs. wind direction and speed for a site using calendarPlot function. However, the wind vectors plotted were completely different in both direction and length when the "statistic" option in the function was set to "mean" and to "max". Can somebody please explain for me why this discrepancy is happening? Isn't the "statistic" option supposed to matter only with pollutants instead of winds? How exactly are the wind vectors calculated in calendarPlot?
![1](https://user-images.githubusercontent.com/37098334/37017293-b3f76e3e-2142-11e8-8472-3a5f5cd23c29.png)
![2](https://user-images.githubusercontent.com/37098334/37017294-b434c054-2142-11e8-934c-67194d4e7b0c.png)

**Comment by davidcarslaw** [comment]
The function will plot the wind direction appropriate to the statistic. So, in this case the wind direction is shown at the time of the daily maximum ozone concentration. I think that would be the expected output?

**Comment by sdonni** [comment]
Hi, i have the same problem, did you have explanations ??
Because i was thinking "The function will plot the wind direction appropriate to the statistic", but with tests i see this is wrong.....so i didn't understand what is the wind vector when the statistic is MAx.

**Comment by davidcarslaw** [comment]
I have pushed a fix for this - please test...

Thanks
David

**Comment by sdonni** [comment]
I've installed devtools, but doesn't change anything on direction wind...

**Comment by davidcarslaw** [comment]
The changes seem to be mostly small. This is what I see for mean and max statistics. I will check again but if you have any data you can share where you see a problem, that would be useful.

`calendarPlot(mydata, pollutant = "o3", year = 2003, annotate = "ws", statistic = "mean")`
<img width="807" alt="image" src="https://github.com/davidcarslaw/openair/assets/1236491/0b301ab4-8bb2-4277-8187-149f7b484ea9">


`calendarPlot(mydata, pollutant = "o3", year = 2003, annotate = "ws", statistic = "max")`
<img width="809" alt="image" src="https://github.com/davidcarslaw/openair/assets/1236491/7f47af4b-dd8f-412f-9d49-8c3286eb4def">

**Comment by sdonni** [comment]
I have a very simple test file with a max PM10 at 12pm.  it can be seen that in the case of statistic=mean,  the direction o f the wind is neither the mean, neither the direction at 12pm (175° ).
![image](https://github.com/davidcarslaw/openair/assets/139781358/1a388730-381d-4196-b197-bc75b556a2c9)
![image](https://github.com/davidcarslaw/openair/assets/139781358/8725011c-376a-416d-afc0-6827804b132a)

**Comment by sdonni** [comment]
Here is my test file.
Version openair: '2.17.0'. Is this ok ??

Thanks.

________________________________
De : David Carslaw ***@***.***>
Envoyé : mercredi 19 juillet 2023 08:01
À : davidcarslaw/openair ***@***.***>
Cc : sdonni ***@***.***>; Comment ***@***.***>
Objet : Re: [davidcarslaw/openair] calendarPlot wind direction and speed (#130)


The changes seem to be mostly small. This is what I see for mean and max statistics. I will check again but if you have any data you can share where you see a problem, that would be useful.

calendarPlot(mydata, pollutant = "o3", year = 2003, annotate = "ws", statistic = "mean")
[image]<https://user-images.githubusercontent.com/1236491/254504850-0b301ab4-8bb2-4277-8187-149f7b484ea9.png>

calendarPlot(mydata, pollutant = "o3", year = 2003, annotate = "ws", statistic = "max")
[image]<https://user-images.githubusercontent.com/1236491/254505003-7f47af4b-dd8f-412f-9d49-8c3286eb4def.png>

**Comment by davidcarslaw** [comment]
No, the version should be as shown (2.17.0.9020). I suggest you restart R and do `devtools::install_github("davidcarslaw/openair")` and try that. Also, I did not get an attachment? Maybe email separately...

**Comment by sdonni** [comment]
Yes version is OK, and wind direction with statistic =max, is OK !!!!!!!    cool, Thanks.

**Comment by davidcarslaw** [comment]
Good! thanks
David
